charts/dapr/charts/dapr_sentry/templates/dapr_sentry_deployment.yaml

```diff
@@ -9,16 +7,16 @@ metadata:
     {{- range $key, $value := .Values.global.k8sLabels }}
     {{ $key }}: {{ tpl $value $ }}
     {{- end }}
-data:
-  {{ if .Values.tls.issuer.certPEM }}issuer.crt: {{ b64enc .Values.tls.issuer.certPEM | trim }}
-  {{ else if $existingSecret }}issuer.crt: {{ index $existingSecret.data "issuer.crt" }}
-  {{ end }}
-  {{ if .Values.tls.issuer.keyPEM }}issuer.key: {{ b64enc .Values.tls.issuer.keyPEM | trim }}
-  {{ else if $existingSecret }}issuer.key: {{ index $existingSecret.data "issuer.key" }}
-  {{end}}
-  {{ if .Values.tls.root.certPEM }}ca.crt: {{ b64enc .Values.tls.root.certPEM | trim }}
-  {{ else if $existingSecret }}ca.crt: {{ index $existingSecret.data "ca.crt" }}
-  {{end}}
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: dapr-trust-bundle
+  labels:
+    app: dapr-sentry
+    {{- range $key, $value := .Values.global.k8sLabels }}
+    {{ $key }}: {{ tpl $value $ }}
+    {{- end }}
```

Comment:
Do we also need to enable encryption or provide an option for that? https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/#configuration-and-determining-whether-encryption-at-rest-is-already-enabled

Reply:
Thanks for the explanation. It helps.
Just for my understanding: So, when we say that Trust Anchor can be accessed from ConfigMap using volume mount, so can any non-Dapr process inside Pod be also able to access this volume and use this Trust Anchor to get itself Authenticated from Sentry?

Reply:
This trust anchor is used by clients (operator, placement, injector) to verify the server (sentry) connection. We use a ConfigMap here because:
1. we don't want any component, apart from Sentry, to be mounting the intermediate certificate and private key,
2. ConfigMap is a better resource type in this scenario because the TrustAnchor is not secret data,
3. In future when we implement other CA signing sources (such as cert-manager), the intermediate Secret may never exist, so using dedicated ConfigMap for the TrustAnchor means that will always exist regardless of the signing/intermediate CA provider.

---

Daprd gets this TrustAnchor from the injector, set as a environment variable, which itself gets it from this ConfigMap. There may be an argument for implementing a controller to distribute the TrustAnchor to all (dapr enabled) namespaces as well so that `daprd`s are able to update their trust store without being restarted (environment variables cannot be updated in place, whereas file mounts can).

---

We likely want to move to using `ClusterTrustBundles` once all supported Kubernetes versions serve this API.

https://github.com/kubernetes/enhancements/issues/3257